src/server/scripts/Shadowlands/SanctumOfDomination/boss_sylvanas_windrunner.cpp

```diff
@@ -36,192 +412,4080 @@ Position const SylvanasIntroPos[4] =
+                else
+                    target->CastSpell(target, SPELL_SYLVANAS_ROOT, CastSpellExtraArgs(TRIGGERED_FULL_MASK).AddSpellMod(SPELLVALUE_DURATION, 1750));
+            }
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectApply += AuraEffectApplyFn(spell_sylvanas_windrunner_disappear::OnApply, EFFECT_1, SPELL_AURA_DUMMY, AURA_EFFECT_HANDLE_REAL);
+    }
+};
+
+// 347928 - Withering Fire
+class spell_sylvanas_windrunner_withering_fire : public SpellScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ spellInfo->GetEffect(EFFECT_0).TriggerSpell });
+    }
+
+    void HandleHit(SpellEffIndex effIndex)
+    {
+        InstanceScript* instance = GetCaster()->GetInstanceScript();
+        if (!instance)
+            return;
+
+        if (Creature* sylvanas = instance->GetCreature(DATA_SYLVANAS_WINDRUNNER))
+        {
+            if (SylvanasEventHelpers::GetWorldStatePhase(GetCaster()) == PHASE_ONE)
+                sylvanas->CastSpell(GetHitUnit(), GetEffectInfo(effIndex).TriggerSpell, true);
+        }
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sylvanas_windrunner_withering_fire::HandleHit, EFFECT_0, SPELL_EFFECT_SCHOOL_DAMAGE);
+    }
+};
+
+// 348627 - Desecrating Shot
+class spell_sylvanas_windrunner_desecrating_shot : public SpellScript
+{
+    bool Validate(SpellInfo const* spellInfo) override
+    {
+        return ValidateSpellInfo({ spellInfo->GetEffect(EFFECT_0).TriggerSpell });
+    }
+
+    void HandleDummyEffect(SpellEffIndex effIndex)
+    {
+        GetCaster()->CastSpell(GetHitUnit(), GetEffectInfo(effIndex).TriggerSpell, true);
+    }
+
+    void Register() override
+    {
+        OnEffectHitTarget += SpellEffectFn(spell_sylvanas_windrunner_desecrating_shot::HandleDummyEffect, EFFECT_0, SPELL_EFFECT_DUMMY);
+    }
+};
+
+// 352663 - Ranger's Heartseeker
+class spell_sylvanas_windrunner_heartseeker : public SpellScript
+{
+    class HeartseekerDamageEvent : public BasicEvent
+    {
+    public:
+        HeartseekerDamageEvent(Unit* actor, Unit* victim) : _actor(actor), _victim(victim) {}
+
+        bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+        {
+            _actor->CastSpell(_victim, SPELL_RANGER_HEARTSEEKER_PHYSICAL_DAMAGE, true);
+            _actor->CastSpell(_victim, SPELL_RANGER_HEARTSEEKER_SHADOW_DAMAGE, true);
+
+            return true;
+        }
+
+    private:
+        Unit* _actor;
+        Unit* _victim;
+    };
+
+    class HeartseekerMissileEvent : public BasicEvent
+    {
+    public:
+        HeartseekerMissileEvent(Unit* actor, Unit* victim) : _actor(actor), _victim(victim) {}
+
+        bool Execute(uint64 /*time*/, uint32 /*diff*/) override
+        {
+            _actor->SendPlaySpellVisual(_victim, SPELL_VISUAL_HEARTSEEKER, 0, 0, 36.0f, false);
```

Comment:
victim might point to invalid memory at this point, only actor is guaranteed to be valid